### PR TITLE
kernel v4.12 support + uppercase fix

### PR DESCRIPTION
**Pull request by kimocoder**
* Added Kernel v4.12 support
* Moved some lowercase to UPPERCASE in device_ids

### CODE REVIEW

File: os_dep/linux/usb_intf.c [hunk 1]
```diff
@@ -164,12 +164,18 @@ static struct usb_device_id rtw_usb_id_tbl[] = {
 	{USB_DEVICE(0x07B8, 0x8812), .driver_info = RTL8812}, /* Abocom - Abocom */
 	{USB_DEVICE(0x2001, 0x3315), .driver_info = RTL8812}, /* D-Link - Cameo */
 	{USB_DEVICE(0x2001, 0x3316), .driver_info = RTL8812}, /* D-Link - Cameo */
-	{USB_DEVICE(0x13b1, 0x003f), .driver_info = RTL8812}, /* Linksys - WUSB6300 */
+	{USB_DEVICE(0x13B1, 0x003F), .driver_info = RTL8812}, /* Linksys - WUSB6300 */
 	{USB_DEVICE(0x2357, 0x0101), .driver_info = RTL8812}, /* TP-Link - Archer T4U */
+	{USB_DEVICE(0x2357, 0x010D), .driver_info = RTL8812}, /* TP-Link - Archer T4U AC1300 */
 	{USB_DEVICE(0x2357, 0x0103), .driver_info = RTL8812}, /* TP-Link - T4UH */
-	{USB_DEVICE(0x20f4, 0x805b), .driver_info = RTL8812}, /* TRENDnet - */
```
Comment (gsra99) [inline os_dep/linux/usb_intf.c]: Could you please say why you have removed TRENDnet from the list of supported devices. This is the device I have.

**Comment by astsam** [comment]
Thank you!
e9eb64d should be reworked. It contains too many duplicated entries.  Comcast - CF-915AC, Netis WF2190, Alfa - AWUS036AC, etc. were supported initially.

**Comment by kimocoder** [comment]
thanks for feedback, wrapped up in a hurry yesterday but I did a cleanup in the new commit

**Comment by StructureBase** [comment]
You pushed different commit before this and it has add virtual interface . Are you gona do it?

**Comment by kimocoder** [comment]
* THIS IS BROKEN

**Pull request closed without merging**